pkg/kubelet/dockertools/manager.go

```diff
@@ -261,24 +273,45 @@ func (self *DockerManager) GetPodStatus(pod *api.Pod) (*api.PodStatus, error) {
 		}
 		dockerContainerName := dockerName.ContainerName
 		c, found := expectedContainers[dockerContainerName]
-		terminationMessagePath := ""
 		if !found {
-			// TODO(dchen1107): should figure out why not continue here
-			// continue
-		} else {
-			terminationMessagePath = c.TerminationMessagePath
+			continue
+		}
+		terminationMessagePath := c.TerminationMessagePath
+		if containerDone.Has(dockerContainerName) {
+			continue
 		}
-		// We assume docker return us a list of containers in time order
+		// We assume docker returns a list of containers sorted in reverse by
+		// time.
 		if containerStatus, found := statuses[dockerContainerName]; found {
-			// Populate last termination state
-			if containerStatus.LastTerminationState.Termination == nil {
-				result := self.inspectContainer(value.ID, dockerContainerName, terminationMessagePath)
-				if result.err == nil && result.status.State.Termination != nil {
-					containerStatus.LastTerminationState = result.status.State
+			// Inspect the container.
+
+			var terminationState *api.ContainerState = nil
+			result := self.inspectContainer(value.ID, dockerContainerName, terminationMessagePath)
```

Comment:
Can we move this inspect outside of the if above? We do it when the if does not trigger as well so we can merge those into one I think.